src/plugins/discover/public/components/field_name/field_name.tsx

```diff
@@ -17,51 +17,32 @@
  * under the License.
  */
 import React from 'react';
-import classNames from 'classnames';
 import { EuiFlexGroup, EuiFlexItem, EuiToolTip } from '@elastic/eui';
 
 import { FieldIcon, FieldIconProps } from '../../../../kibana_react/public';
 import { shortenDottedString } from '../../helpers';
 import { getFieldTypeName } from './field_type_name';
 
-// property field is provided at discover's field chooser
 // properties fieldType and fieldName are provided in kbn_doc_view
 // this should be changed when both components are deangularized
 interface Props {
-  field?: {
-    type: string;
-    name: string;
-    rowCount?: number;
-    scripted?: boolean;
-  };
-  fieldName?: string;
-  fieldType?: string;
+  fieldName: string;
+  fieldType: string;
   useShortDots?: boolean;
   fieldIconProps?: Omit<FieldIconProps, 'type'>;
 }
 
-export function FieldName({ field, fieldName, fieldType, useShortDots, fieldIconProps }: Props) {
-  const type = field ? String(field.type) : String(fieldType);
+export function FieldName({ fieldName, fieldType, useShortDots, fieldIconProps }: Props) {
+  const type = String(fieldType);
   const typeName = getFieldTypeName(type);
 
-  const name = field ? String(field.name) : String(fieldName);
+  const name = String(fieldName);
   const displayName = useShortDots ? shortenDottedString(name) : name;
 
-  const noResults = field ? !field.rowCount && !field.scripted : false;
-
-  const className = classNames('dscFieldName', {
-    'dscFieldName--noResults': noResults,
-  });
-
   return (
-    <EuiFlexGroup className={className} alignItems="center" gutterSize="s" responsive={false}>
+    <EuiFlexGroup alignItems="center" gutterSize="s" responsive={false}>
       <EuiFlexItem grow={false}>
-        <FieldIcon
-          type={type}
-          label={typeName}
-          scripted={field ? field.scripted : false}
-          {...fieldIconProps}
-        />
+        <FieldIcon type={type} label={typeName} scripted={false} {...fieldIconProps} />
```

Comment:
Could you please comment here what's happened with scripted?
Is it a legacy functionality?

Reply:
excellent question, seems kind of legacy and unused, changed that:

<img width="232" alt="Bildschirmfoto 2020-03-27 um 10 50 06" src="https://user-images.githubusercontent.com/463851/77743769-d7301500-7018-11ea-9a62-a32cb55c929b.png">